src/components/ContactCard/ContactForm.jsx

```diff
@@ -77,7 +77,62 @@ const initialFieldValues = fields.reduce((initialValues, { name, isArray }) => {
   return initialValues
 }, {})
 
-class ContactForm extends React.Component {
+export const moveToHead = shouldBeHead => items =>
+  items.reduce((arr, v) => (shouldBeHead(v) ? [v, ...arr] : [...arr, v]), [])
+
+const movePrimaryToHead = moveToHead(v => v.primary)
+
+export class DumbContactForm extends React.Component {
+  getInitialValues = () => {
+    const { contact } = this.props
+    if (contact) {
+      const {
+        address,
+        birthday,
+        company,
+        cozy,
+        email,
+        name,
+        note,
+        phone
+      } = contact
+      const addressValue = address
+        ? movePrimaryToHead(address).map(({ formattedAddress, type }) => ({
+            address: formattedAddress,
+            addressLabel: type
+          }))
+        : [undefined]
+      const cozyValue = cozy ? cozy[0].url : undefined
+      const cozyLabel = cozy && cozy[0].label ? cozy[0].label : undefined
+      const emailValue = email
+        ? movePrimaryToHead(email).map(item => ({
+            email: item.address,
+            emailLabel: item.type
+          }))
+        : [undefined]
+      const phoneValue = phone
+        ? movePrimaryToHead(phone).map(item => ({
+            phone: item.number,
+            phoneLabel: item.type
+          }))
+        : [undefined]
+      return {
+        givenName: name.givenName,
+        familyName: name.familyName,
+        phone: phoneValue,
+        email: emailValue,
+        address: addressValue,
+        cozy: cozyValue,
+        cozyLabel: cozyLabel,
+        company,
+        birthday,
+        note
+      }
+    }
+
+    return initialFieldValues
+  }
```

Comment:
Can we move this method outside of the component since it has nothing to do with the rendering, just formating.

Reply:
I agree but if I do this I should probably also move `formDataToContact` method outside too (rewrite it so it only convert data and add a `handleSubmit` method that calls this function and `onSubmit(contact)`). For the react-final-form forms, I usually used a folder for the form with the component and various utils in separate files (`load.js`, `save.js`, `validate.js`)

Reply:
Like the idea

Reply:
@Crash-- you talk about the `save.js`/`load.js/`validate.js` files?
@y-lohse what do you think about it?

Reply:
👍 for separating the purely logic-code into standalone files. And as long as the code is coupled to the forms, I think having them next to the components is good, so `save.js/load.js` works for me too!

Reply:
In the end, I chose `contactToFormValues.js` and `formValuesToContact.js` because load and save suggest that the data is fetched and pushed to the server